### ISSUE TITLE
FR: Themes available from a token into URLs

### ISSUE THREAD

**Issue by cara-tm** [textpattern/textpattern#1787]
### Is your feature request related to a problem?

No!

### What is the feature?

I don't know if it's possible: it will be very useful to have a kind of token added into the URL to serve a definite theme (maybe governed via "Preferences"). That feature could allow anybody to test a new website model (for clients, for online tools, for SmartPhones checking, etc.)

**Comment by Bloke** [comment]
I'm confused. Setting a dev theme automatically shows the in-dev experience to anyone logged in. Usually, letting your client have a managing editor login allows them to see what's going on, and they can sign the changes off ready for go-live. If they sign in on different devices, that's the experience they see.

So you want some URL patam to deliver the in-dev experience _without_ being logged in?

**Comment by phiw13** [comment]
@Bloke —Use case: I am (or the client / stakeholder / … is) logged in on a desktop computer and want to view the result on a small screen device. All the while editing the page(s) (text/element placement/design/…).

**Comment by Bloke** [comment]
@phiw13 Yeah, that would be more handy than having to log in on a second device, which kicks you out of the desktop. I usually get round it with a pair of accounts. Not ideal.

I can't think of a simple way to do this, besides assigning each theme some unique hash (somehow) so you can tack that onto the URL and have it render the dev theme instead of the live one. It would have to be something unguessable, otherwise anybody could potentially see it.

**Comment by cara-tm** [comment]
@Bloke

No worries. Just an idea. The reason why I said I don't think it's possible.

**Comment by cara-tm** [comment]
I tried but it doesn't seem to work...

**Comment by petecooper** [comment]
>I can't think of a simple way to do this, besides assigning each theme some unique hash (somehow) so you can tack that onto the URL and have it render the dev theme instead of the live one. It would have to be something unguessable, otherwise anybody could potentially see it.

Not currently viable – closing for now.

**Comment by Bloke** [comment]
As an aside, this is now less of an issue in 4.9.0 if you list the account login names in the Advanced Pref for _Concurrent accounts_. That will allow your client to login on multiple devices to see and edit the dev theme.
